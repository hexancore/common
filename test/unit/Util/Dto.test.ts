import { OtherTestDTO, TestDTO, TestValueObject } from "@test/helper/TestDto";

/**
 * @group unit
 */

describe("Dto", () => {
  test("toJSON", () => {
    const dto = TestDTO.cs({
      booleanField: true,
      numberArrayField: [1, 2],
      numberField: 1,
      stringField: "test",
      optionalDtoArrayField: [OtherTestDTO.cs({ primitiveField: 1 })],
      optionalDtoField: OtherTestDTO.cs({ primitiveField: 2 }),
      bigIntField: 1000n,
      optionalValueObjectField: TestValueObject.cs(2),
      optionalValueObjectArrayField: [TestValueObject.cs(3), TestValueObject.cs(4)],
    });

    const current = dto.toJSON();

    expect(current).toMatchSnapshot();
  });
});
