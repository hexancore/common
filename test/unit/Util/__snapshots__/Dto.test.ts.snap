// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dto toJSON 1`] = `
{
  "bigIntField": "1000",
  "booleanField": true,
  "numberArrayField": [
    1,
    2,
  ],
  "numberField": 1,
  "optionalDtoArrayField": [
    {
      "primitiveField": 1,
    },
  ],
  "optionalDtoField": {
    "primitiveField": 2,
  },
  "optionalStringField": undefined,
  "optionalValueObjectArrayField": [
    3,
    4,
  ],
  "optionalValueObjectField": 2,
  "stringField": "test",
}
`;
